### PR TITLE
✏️ Better HTML conversion options

### PR DESCRIPTION
**Pull request by asim-shrestha**
> [!IMPORTANT]
> Adds HTML conversion feature with markdown and text options, including error handling and tests.
> 
>   - **Behavior**:
>     - Adds `html_converter_type` parameter to `capture_html()` in `core.py` for specifying conversion type.
>     - Raises `UnknownHTMLConverter` error for unsupported types in `get_html_converter()`.
>   - **HTML Conversion**:
>     - Introduces `HTMLToMarkdownConverter` and `HTMLToTextConverter` for HTML to markdown and text conversion.
>     - Implements `get_html_converter()` in `html_converter/__init__.py` to select converter based on type.
>   - **Errors**:
>     - Adds `UnknownHTMLConverter` exception in `errors.py` for handling unknown types.
>   - **Tests**:
>     - Adds `test_capture_html_conversion_types()` in `test_e2e.py` to verify conversion to markdown and text.
>     - Includes mock HTML files `heading.html` and `table.html` for testing.
>   - **Misc**:
>     - Updates version to `0.53.0` in `pyproject.toml` and `uv.lock`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=reworkd%2Fharambe&utm_source=github&utm_medium=referral)<sup> for ed3ae6b03cf9991fdad047c298977f0e642bcfac. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: sdk/harambe/html_converter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+from sdk.harambe.html_converter.html_to_markdown import HTMLToMarkdownConverter
+from sdk.harambe.html_converter.html_to_text import HTMLToTextConverter
+
+HTMLConverterType = Literal["markdown", "text"]
+
+
+def get_html_converter(
+    html_converter_type: HTMLConverterType | None,
+) -> MarkdownConverter:
```
[the remark this answers is not in the document]
Comment (asim-shrestha) [reply sdk/harambe/html_converter/__init__.py:13]: no

File: core/harambe_core/errors.py [hunk 2]
```diff
@@ -4,6 +4,13 @@ class HarambeException(Exception):
     pass
 
 
+class UnknownHTMLConverter(HarambeException):
+    def __init__(self, converter_type: any) -> None:
```
[the remark this answers is not in the document]
Comment (mohamedmamdouh22) [reply core/harambe_core/errors.py:8]: ellipsis is right, it should be `Any` from the typing module

File: sdk/harambe/html_converter/__init__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    if html_converter_type == "markdown":
+        return HTMLToMarkdownConverter()
+    if html_converter_type == "text":
+        return HTMLToTextConverter()
+    else:
+        raise UnknownHTMLConverter(html_converter_type)
```
Comment (awtkns) [inline sdk/harambe/html_converter/__init__.py:19]: remove else

File: core/harambe_core/errors.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (asim-shrestha) [reply core/harambe_core/errors.py]: good call

**Pull request merged**